pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -464,6 +740,101 @@ func TestDropStatefulSetDisabledFields(t *testing.T) {
+			enableMaxUnavailable: true,
+			ss: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(15),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 2},
+				false,
+			),
+			oldSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(15),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 2},
+				false,
+			),
+			expectedSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(15),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 2},
+				false,
+			),
+		},
+		{
+			name:                 "zero minReadySeconds with maxUnavailable enabled",
+			enableMaxUnavailable: true,
+			ss: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(0),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+				false,
+			),
+			oldSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(0),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+				false,
+			),
+			expectedSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(0),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+				false,
+			),
+		},
+		{
+			name:                 "minReadySeconds change with maxUnavailable enabled",
+			enableMaxUnavailable: true,
+			ss: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(45),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+				false,
+			),
+			oldSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+				false,
+			),
+			expectedSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(45),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+				false,
+			),
+		},
+		{
+			name:                 "minReadySeconds disabled with maxUnavailable disabled",
+			enableMaxUnavailable: false,
+			ss: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 2},
+				false,
+			),
+			oldSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				nil,
+				false,
+			),
+			expectedSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				nil,
+				true,
+			),
+		},
+		{
+			name:                 "transition from disabled to enabled maxUnavailable with minReadySeconds",
+			enableMaxUnavailable: true,
+			ss: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 2},
+				false,
+			),
+			oldSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				nil,
+				false,
+			),
+			expectedSS: makeStatefulSetWithMaxUnavailable(
+				generateStatefulSetWithMinReadySeconds(30),
+				&intstr.IntOrString{Type: intstr.Int, IntVal: 2},
+				false,
+			),
+		},
```

Comment:
After carefully reviewing this entire test I think after I [cleaned up the old tests](https://github.com/kubernetes/kubernetes/pull/133681) there's no need to introduce any new tests here. We have sufficient coverage for MaxUnavailable here, covering enabled and disabled feature leaving and dropping the field as required.